### DISCUSSION TITLE
How to fetch all topics

### DISCUSSION THREAD

**Discussion by tareefdev** [magit/forge#599] [Support] [answered]
How to set up Forge so it pulls all topics from a repo. Here is my current config:

````lisp
  (use-package forge
  :after magit
  :config
  (setq auth-sources '("~/.authinfo"))
  (setq forge-topic-list-limit  -1)))
````
`forge-topic-list-limit` seems to control the number of listed topics, not the number of pulled topics. For example, I tried `(setq forge-topic-list-limit  '(300 . -1))`, but forge did not pull all open issues in my repo (112 ones).

**Comment by tarsius** [comment]
Forge should fetch all topics without any configuration. How did you determine that it didn't do that?

**Reply by tareefdev** [reply]
````bash
sqlite3 ~/.emacs.d/forge-database.sqlite "SELECT COUNT(*) FROM issue WHERE repository = '\"Z2l0bGFiLmNvbToyMzc0MTEyNg==\"';"
103
````
While this repo contains 116 open issues and 629 closed issues.

**Reply by tareefdev** [reply]
Also, I tried to reset the database.

**Reply by tarsius** [reply]
I think it is a bug in the Github API implementation. It promises to return all topics (with pagination, but still) but then it randomly decides to not do that when it doesn't feel like it. (Recently having had to write [this](https://github.com/magit/forge/blob/notif/docs/forge.org#dealing-with-githubs-abysmal-notification-api), I am more inclined to believe it is a mistake on their end than on mine.)

If you tell me for what repository this happens for you, I can try to reproduce.

**Reply by tarsius** [reply]
On thing that might help is to reduce `forge.graphqlItemLimit` to 50 or so, which can be done from `forge-dispatch`. You will have to remove the repository in question from the database, before doing that.

**Reply by tareefdev** [reply]
I need to remember to mention that all my repos are on GitLab. I did not manage to utilize `forge-dispatch` to set `forge.graphqlItemLimit` to 50, but I read [here](https://github.com/magit/forge/issues/69#issuecomment-1572402440) that setting `(setq ghub-graphql-items-per-request 50)` is another way to have it applied globally. After resetting the database, I tried to pull topics of [this repo](https://gitlab.com/prism-break/prism-break), but again, only 100 topics came.

**Reply by tarsius** [reply, marked as answer]
Fixed in https://github.com/magit/ghub/commit/0d7c81eee3ba0c6e029605e545173721c39947d7.

Thanks for the report!

**Reply by tareefdev** [reply]
Thanks for your great efforts!
